addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -273,229 +281,241 @@ export class LinkPlugin extends Plugin {
         };
         return pasteAsURLCommand;
     }
+
     /**
-     * Toggle the Link popover to edit links
+     * open the Link popover to edit links
      *
-     * @param {Object} options
-     * @param {HTMLElement} options.link
+     * @param {HTMLElement} [linkElement]
      */
-    toggleLinkTools({ link } = {}) {
-        if (!link) {
-            link = this.getOrCreateLink();
+    openLinkTools(linkElement) {
+        this.closeLinkTools();
+        let selection = this.dependencies.selection.getEditableSelection();
+        let cursorsToRestore = this.dependencies.selection.preserveSelection();
+        const commonAncestor = closestElement(selection.commonAncestorContainer);
+        linkElement = linkElement || findInSelection(selection, "a");
+        if (linkElement) {
+            const linkWasExtended = this.extendsLinkToSelection(linkElement, selection);
+            if (linkWasExtended) {
+                linkElement = findInSelection(selection, "a");
+                this.dependencies.history.addStep();
+                cursorsToRestore = this.dependencies.selection.preserveSelection();
+            }
         }
-        this.linkElement = link;
-    }
-
-    normalizeLink() {
-        const { anchorNode } = this.dependencies.selection.getEditableSelection();
-        const linkEl = closestElement(anchorNode, "a");
-        if (linkEl && linkEl.isContentEditable) {
-            const label = linkEl.innerText;
-            const url = deduceURLfromText(label, linkEl);
-            if (url) {
-                linkEl.setAttribute("href", url);
+        this.linkInDocument = linkElement;
+        if (!linkElement) {
+            // create a new link element
+            linkElement = this.document.createElement("a");
+            if (!selection.isCollapsed) {
+                linkElement.append(selection.textContent());
             }
         }
-    }
 
-    handleSelectionChange(selectionData) {
-        const selection = selectionData.editableSelection;
+        const selectionTextContent = selection?.textContent();
+        const isImage = !!findInSelection(selection, "img");
+
+        const applyCallback = (url, label, classes) => {
+            if (this.linkInDocument && isImage) {
+                if (url) {
+                    this.linkInDocument.href = url;
+                } else {
+                    this.linkInDocument.removeAttribute("href");
+                }
+            } else if (this.linkInDocument) {
+                if (url) {
+                    this.linkInDocument.href = url;
+                } else {
+                    this.linkInDocument.removeAttribute("href");
+                }
```

Comment:
Nit picking, I think we can use `this.linkInDocument.href = url;` without checking the `url`. If the href is `""`, the href is used as the one of current page so we kinda avoid a link with no url. 

reproduce: 
1. create a normal link with url
2. delete the url in the popover and apply 
3. a link without url

Reply:
This is a spec change we discussed with Lionel.
The decision was made to remove the href altogether when the url is removed by the user.
This change the behaviors of the link popover to be directly in edit mode when the selection is in a link without url.